rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -57,16 +57,16 @@ pub type ExecutionResult = Result<Executed, ExecutionError>;
 
 impl LightFetch {
 	/// Get a block header from the on demand service or client, or error.
-	pub fn header(&self, id: BlockId) -> BoxFuture<Option<encoded::Header>, Error> {
+	pub fn header(&self, id: BlockId) -> BoxFuture<encoded::Header, Error> {
```

Comment:
yep, although I think optimizing this stuff is beyond the scope of the PR. Most of these RPC helpers are redundant with batch requests after #5573